ReactiveCocoa/UIKit/iOS/UIKeyboard.swift

```diff
@@ -37,19 +68,43 @@ public struct KeyboardChangeContext {
 		return base[UIKeyboardIsLocalUserInfoKey] as! Bool
 	}
 
-	fileprivate init(_ userInfo: [AnyHashable: Any]) {
+	fileprivate init(userInfo: [AnyHashable: Any], event: KeyboardEvent) {
 		base = userInfo
+		self.event = event
 	}
 }
 
 extension Reactive where Base: NotificationCenter {
+	/// Create a `Signal` that notifies whenever the system keyboard announce specified event.
+	///
+	/// - parameters:
+	///   - event:  The type of system keyboard event to observe.
+	///
+	/// - returns: A `Signal` that emits the context of system keyboard event.
+	public func keyboard(_ event: KeyboardEvent) -> Signal<KeyboardChangeContext, NoError> {
+		return notifications(forName: event.notificationName)
+			.map { notification in KeyboardChangeContext(userInfo: notification.userInfo!, event: event) }
+	}
+	
+	/// Create a `Signal` that notifies whenever the system keyboard announces specified events.
+	///
+	/// - parameters:
+	///   - first: First type of system keyboard event to observe.
+	///   - second: Second type of system keyboard event to observe.
+	///   - tail: Rest of the types of system keyboard events to observe.
+	///
+	/// - returns: A `Signal` that emits the context of system keyboard events.
+	public func keyboard(_ first: KeyboardEvent, _ second: KeyboardEvent, _ tail: KeyboardEvent...) -> Signal<KeyboardChangeContext, NoError> {
```

Comment:
Not sure why we would need two code paths. Variadic supports 1 argument.

Reply:
@andersio
Because it was commented as above,  I thought that it's better not to `Signal.merge` when only 1 argument.
Two code paths are techniques to keep arguments from being empty.
Variadic allows the codes such as below.
```swift
NotificationCenter.default.reactive.keyboard()
    .observeValues {...}
```
If better to allow them, I'll fix it.